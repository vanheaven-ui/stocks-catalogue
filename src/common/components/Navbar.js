import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="navbar" style={{ color: '#007bff' }}>
    <h2><Link to="/">StocksCatalogue</Link></h2>
    <div className="links">
      <Link to="/" style={{ margin: '0 10px 0 0' }}>Home</Link>
      <Link to="/news">Stock News</Link>
    </div>
  </nav>
);

export default Navbar;
