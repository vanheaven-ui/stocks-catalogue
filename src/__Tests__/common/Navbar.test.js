import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react';
import Navbar from '../../common/components/Navbar';
import rootNode from '../helpers';
import '@testing-library/jest-dom/extend-expect';

describe('Navbar component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    ReactDOM.render(
      <BrowserRouter><Navbar /></BrowserRouter>,
      rootNode(),
    );
  });

  test('doesnot change unexpectedly', () => {
    const navbarTree = renderer.create(
      <BrowserRouter><Navbar /></BrowserRouter>,
    ).toJSON();

    expect(navbarTree).toMatchSnapshot();
  });

  test('renders into the document', () => {
    const { getByText } = render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(getByText(/StocksCatalogue/)).toBeInTheDocument();
    expect(getByText(/Home/)).toBeInTheDocument();
    expect(getByText(/StockNews/)).toBeInTheDocument();
  });

  test('elements have right attributes', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText(/StocksCatalogue/)).toHaveAttribute('href', '/');
    expect(screen.getByText(/Home/)).toHaveAttribute('href', '/');
    expect(screen.getByText(/StockNews/)).toHaveAttribute('href', '/news');
  });
});
