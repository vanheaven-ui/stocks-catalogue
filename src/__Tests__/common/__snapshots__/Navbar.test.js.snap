// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component doesnot change unexpectedly 1`] = `
<nav
  className="navbar sticky-top mb-2"
  style={
    Object {
      "color": "#007bff",
    }
  }
>
  <h2>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="my logo"
        src="logo.jpeg"
        style={
          Object {
            "height": 60,
            "marginRight": 5,
            "width": 60,
          }
        }
      />
      StocksCatalogue
    </a>
  </h2>
  <div
    className="links"
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "margin": "0 10px 0 0",
        }
      }
    >
      Home
    </a>
    <a
      href="/news"
      onClick={[Function]}
    >
      StockNews
    </a>
  </div>
</nav>
`;
